### PR TITLE
doc: Clarify that `DisplayServer.window_set_*_callback` aren't supported on Window nodes

### LINKED ISSUE #81830: Mouse Events weird when using DisplayServer.window_set_window_event_callback
### Godot version

4.2 dev 4

### System information

Godot v4.2.dev4 - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3070 (NVIDIA; 31.0.15.3699) - 12th Gen Intel(R) Core(TM) i7-12700KF (20 Threads)

### Issue description

I have an issue that starts from 4.2 dev 4 and up where the mouse_entered/exited events just not trigger and after a bit of initial confusion I understand it comes from using `DisplayServer.window_set_window_event_callback`.

Now I might just be using that wrong but yeah basically it seems like you can't use your own callback anymore, I'm not sure. 

It might be related to #80334 ?

### Steps to reproduce

Check the script and toggle line 4.

### Minimal reproduction project

[mrp.zip](https://github.com/godotengine/godot/files/12643757/mrp.zip)

### PR DESCRIPTION
**Pull request by akien-mga**
- Alternative to #84638, see that PR for details.
- Closes #81830.

### CODE REVIEW

File: doc/classes/DisplayServer.xml [hunk 1]
```diff
@@ -1403,6 +1403,7 @@
 			<param index="1" name="window_id" type="int" default="0" />
 			<description>
 				Sets the [param callback] that should be called when files are dropped from the operating system's file manager to the window specified by [param window_id].
+				[b]Note:[/b] Advanced users only! Adding such a callback to a [Window] node will override its default implementation, which can introduce bugs.
```
Comment (YuriSizov) [inline doc/classes/DisplayServer.xml:1406]:
```suggestion
				[b]Warning:[/b] Advanced users only! Adding such a callback to a [Window] node will override its default implementation, which can introduce bugs.
```

Perhaps?

Comment (akien-mga) [reply doc/classes/DisplayServer.xml]: Done.

**Review by deralmas — APPROVED** [review]
This addresses my concern about API compatibility, nice!

Also yeah, the other `window_set_*_callback` stuff is definitely useful if you know what you're doing. I don't think that it'd be worth removing them, so adding a disclaimer is probably the best approach all things considered.

**Review by Sauermann — APPROVED** [review]
Looking good to me.
Non-breaking changes are always better than breaking changes: supersedes #84638

**Pull request merged**